### PR TITLE
[v13] Update docker images

### PR DESCRIPTION
**Pull request by jakule**
Backport of https://github.com/gravitational/teleport/pull/27175  and https://github.com/gravitational/teleport/pull/26905

### CODE REVIEW

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -96,7 +94,7 @@ RUN mkdir -p /opt && cd /opt &&  \
 # 4. Fast, language-dependent dependencies
 # 5. Multi-stage layer copies
 
-FROM ubuntu:18.04 AS buildbox
+FROM ubuntu:20.04 AS buildbox
```
Comment (r0mant) [inline build.assets/Dockerfile:97]: I might have missed it in the original discussion, but wouldn't bumping version to 20.04 raise our minimum glibc version requirement?

**Review by r0mant — CHANGES_REQUESTED** [review]
Requesting changes until we understand how this affects our glibc compatibility.

Moreover, I'm not sure if this should be backported to stable release branches at all tbh to avoid breaking compatibility mid-release. We've been there before and it generates support load. Is there a reason we'd want to backport?

File: build.assets/Dockerfile (hunk 1 above)
Comment (jakule) [reply build.assets/Dockerfile:97]: I want to backport this as Ubuntu 18.04 is EOL. Bumping the glibc version on the buildbox won't change the glibc version required by Teleport as we're using CentOS 7 for Teleport and the new Connect image for Teleport Connect releases.

**Review by ravicious — APPROVED** [review]

**Review by codingllama — APPROVED** [review]
Please make sure @r0mant is happy with the changes before merging this or the other backports, otherwise LGTM.

**Review by gzdunek — APPROVED** [review]
This PR only raises the glibc requirement of Teleport Connect to 2.28 as it is built on Debian 10.

**Review by r0mant — APPROVED** [review]

**Pull request merged**